File: backend/app/schemas/background.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime


class ExperienceCreate(BaseModel):
    company: str
    role: str
    duration: str
    description: str


class ExperienceResponse(ExperienceCreate):
    id: str
    created_at: datetime


class STARStoryCreate(BaseModel):
    title: str
    situation: str
    task: str
    action: str
    result: str
    tags: List[str] = []


class STARStoryResponse(BaseModel):
    id: str
    title: str
    situation: str
    task: str
    action: str
    result: str
    tags: List[str]
    created_at: datetime


class BackgroundCreate(BaseModel):
    name: str
    resume_text: Optional[str] = None
    skills: List[str] = []


class BackgroundUpdate(BaseModel):
    name: Optional[str] = None
    resume_text: Optional[str] = None
    skills: Optional[List[str]] = None


class BackgroundResponse(BaseModel):
    id: str
    name: str
    resume_text: Optional[str]
    skills: List[str]
    experiences: List[ExperienceResponse] = []
    star_stories: List[STARStoryResponse] = []
    created_at: datetime
    updated_at: datetime


# Resume parsing schemas
class ParsedExperienceResponse(BaseModel):
    company: str
    role: str
    duration: str
    description: str


class ParsedSTARStoryResponse(BaseModel):
    title: str
    situation: str
    task: str
    action: str
    result: str
    tags: List[str] = []


class ResumeParseRequest(BaseModel):
    resume_text: str


class ResumeParseResponse(BaseModel):
    name: str
    skills: List[str]
    experiences: List[ParsedExperienceResponse]
    star_stories: List[ParsedSTARStoryResponse]

File: frontend/src/types/background.ts
export interface Experience {
  id: string;
  company: string;
  role: string;
  duration: string;
  description: string;
  created_at: string;
}

export interface STARStory {
  id: string;
  title: string;
  situation: string;
  task: string;
  action: string;
  result: string;
  tags: string[];
  created_at: string;
}

export interface Background {
  id: string;
  name: string;
  resume_text?: string;
  skills: string[];
  experiences: Experience[];
  star_stories: STARStory[];
  created_at: string;
  updated_at: string;
}

export interface BackgroundCreate {
  name: string;
  resume_text?: string;
  skills: string[];
}

export interface STARStoryCreate {
  title: string;
  situation: string;
  task: string;
  action: string;
  result: string;
  tags: string[];
}

export interface ExperienceCreate {
  company: string;
  role: string;
  duration: string;
  description: string;
}

// Resume parsing types
export interface ParsedExperience {
  company: string;
  role: string;
  duration: string;
  description: string;
}

export interface ParsedSTARStory {
  title: string;
  situation: string;
  task: string;
  action: string;
  result: string;
  tags: string[];
}

export interface ResumeParseResponse {
  name: string;
  skills: string[];
  experiences: ParsedExperience[];
  star_stories: ParsedSTARStory[];
}
